packages/jwk-did-provider/src/jwk-did-provider.ts

```diff
@@ -0,0 +1,119 @@
+import { IIdentifier, IKey } from '@veramo/core'
+import { AbstractIdentifierProvider } from '@veramo/did-manager'
+import base64url from 'base64url'
+import * as jose from 'jose'
+import { generatePrivateKeyHex } from '../src/functions'
+import {
+  IAddKeyArgs,
+  IAddServiceArgs,
+  ICreateIdentifierArgs,
+  IImportProvidedOrGeneratedKeyArgs,
+  IRemoveKeyArgs,
+  IRequiredContext,
+  KeyType,
+  KeyUse,
+} from './types/jwk-provider-types'
+import Debug from 'debug'
+
+const debug = Debug('veramo:jwk-did-provider')
+
+/**
+ * {@link @veramo/did-manager#DIDManager} identifier provider for `did:ion` identifiers
+ * @public
+ */
+export class JwkDIDProvider extends AbstractIdentifierProvider {
+  private readonly defaultKms: string
+
+  constructor(options: { defaultKms: string }) {
+    super()
+    this.defaultKms = options.defaultKms
+  }
+
+  /** {@inheritDoc @veramo/veramo-core#IDIDManager.didManagerCreate} */
+  async createIdentifier(args: ICreateIdentifierArgs, context: IRequiredContext): Promise<Omit<IIdentifier, 'provider'>> {
+    const key = await this.importProvidedOrGeneratedKey(
+      {
+        kms: args.kms,
+        options: args.options,
+      },
+      context
+    )
+    const jwk = await jose.exportJWK(Uint8Array.from(Buffer.from(key.publicKeyHex, 'hex')))
+
```

Comment:
Probably wise to inspect the JWK for a private key being present here.

According to the [DID:JWK spec](https://github.com/quartzjer/did-jwk/blob/main/spec.md#security):
> Only JWKs containing public key material may be used, a JWK for a private key must never be used and must be rejected by all implementations when encountered.

Reply:
i am not sure about this one. the jwk we make here is always only from the public key. This check would be something for the resolver?